src/libraries/Native/Unix/System.Globalization.Native/pal_localeStringData.c

```diff
@@ -221,28 +224,59 @@ int32_t GlobalizationNative_GetLocaleInfoString(const UChar* localeName,
     switch (localeStringData)
     {
         case LocaleString_LocalizedDisplayName:
-            uloc_getDisplayName(locale, DetectDefaultLocaleName(), value, valueLength, &status);
+            assert(uiLocaleName != NULL);
+            GetLocale(uiLocaleName, uiLocale, ULOC_FULLNAME_CAPACITY, false, &status);
+            uloc_getDisplayName(locale, uiLocale, value, valueLength, &status);
+            if (status == U_USING_DEFAULT_WARNING)
+            {
+                // The default locale data was used. i.e. couldn't find suitable resources for the requested UI language. Fallback to English then.
+                uloc_getDisplayName(locale, ULOC_ENGLISH, value, valueLength, &status);
+            }
             break;
         case LocaleString_EnglishDisplayName:
             uloc_getDisplayName(locale, ULOC_ENGLISH, value, valueLength, &status);
             break;
         case LocaleString_NativeDisplayName:
             uloc_getDisplayName(locale, locale, value, valueLength, &status);
+            if (status == U_USING_DEFAULT_WARNING)
+            {
+                // The default locale data was used. i.e. couldn't find suitable resources for the requested input locale. Fallback to English instead.
+                uloc_getDisplayName(locale, ULOC_ENGLISH, value, valueLength, &status);
+            }
+
             break;
         case LocaleString_LocalizedLanguageName:
-            uloc_getDisplayLanguage(locale, DetectDefaultLocaleName(), value, valueLength, &status);
```

Comment:
Is the `DetectDefaultLocaleName()` method still used anywhere? Can it be removed now?

Reply:
This still needed here 

https://github.com/dotnet/runtime/blob/190c69d78f42f6d312d66e3676eef3a714f7bdf2/src/libraries/Native/Unix/System.Globalization.Native/pal_locale.c#L217